openwisp_controller/config/base/vpn.py

```diff
@@ -541,8 +567,11 @@ def _get_common_name(self):
 
     @classmethod
     def post_save(cls, instance, **kwargs):
-        # regenerates peer cache
-        instance.vpn._invalidate_peer_cache(update=True)
+        def _post_save():
+            instance.vpn._invalidate_peer_cache(update=True)
+            instance.vpn.update_vpn_server_configuration()
+
+        transaction.on_commit(_post_save)
```

Comment:
@nemesisdesign can you please double check this change? 

I did these because `vpn._invalidate_peer_cache` makes database queries and it might be possible that it gets triggered before the database transaction for saving VPNClient is commited. Is my concern correct?

Reply:
looks good :+1: